src/main/java/org/springframework/data/couchbase/core/ReactiveFindByQueryOperationSupport.java

```diff
@@ -119,20 +127,43 @@ public ReactiveFindByQueryOperationSupport(final ReactiveCouchbaseTemplate templ
 		private String assembleEntityQuery(final boolean count) {
 			final String bucket = "`" + template.getBucketName() + "`";
 
-			final StringBuilder statement = new StringBuilder("SELECT ");
-			if (count) {
-				statement.append("count(*) as __count");
-			} else {
-				statement.append("meta().id as __id, meta().cas as __cas, ").append(bucket).append(".*");
-			}
+			final StringBuilder statement = new StringBuilder();
 
-			statement.append(" FROM ").append(bucket);
+			if(!query.hasInlineN1qlQuery()) { // the strings below could come from StringN1qlQueryParser
+				statement.append("SELECT ");
+				if (count) {
+					statement.append("count(*) as __count");
+				} else {
+					statement.append("meta().id as __id, meta().cas as __cas, ").append(bucket).append(".*");
```

Comment:
the template utils __id and so forth should be used here too?

Reply:
I fixed this to use the bits produced from StringN1qlQueryParser